js/filter-set.js

```diff
@@ -64,9 +63,9 @@ FilterSet.prototype.clear = function() {
   });
 };
 
-FilterSet.prototype.handleTagRemove = function(e) {
-  var $input = this.$body.find('#' + e.key);
-  var type = $input.attr('type');
+FilterSet.prototype.handleTagRemove = function(e, opts) {
+  var $input = this.$body.find('#' + opts.key);
+  var type = $input.attr('type') || 'text';
```

Comment:
Actually, I'm proposing a little change that handles this a bit better. It turns out that `$input.attr('type')` gives the value of the `type` as encoded in the source, which means it's `undefined` if we didn't provide a `type` for some element. But using the native DOM interface works here, since it gives us the type that the element has taken on, either because it was set explicitly or because it was the default value: `$input.get(0).type`.